docs/webhook.asciidoc

```diff
@@ -0,0 +1,191 @@
+[id="{p}-webhook"]
+== Validating webhook
+
+A validating webhook provides additional validation of Elasticsearch resources beyond what can be specified in OpenAPI specs.
+This allows advanced validation to help prevent Elasticsearch resources that are invalid and will not function correctly
+from being admitted by the Kubernetes API server, which can aid troubleshooting.
```

Comment:
I would go maybe for a bit more general introduction something along the lines of: validating webhooks give you immediate feedback on the Elasticsearch manifests you submit and allow you to catch errors right away before ECK even tries to fullfill your request.